src/datasets/utils/download_manager.py

```diff
@@ -182,6 +183,10 @@ def download(self, url_or_urls):
             downloaded_path(s): `str`, The downloaded paths matching the given input
                 url_or_urls.
         """
+        if self._download_config.splits:
+            if isinstance(url_or_urls, dict) and all(split in url_or_urls for split in self._download_config.splits):
```

Comment:
I would add an additional check just to avoid unwanted behaviors.
For example right now if a dataset script passes to the dl_manager a dict like this:
```
{"main_data": url_to_main_data, "secondary_data": url_to_sec_data}
```
then this trick here would use `url_or_urls = {}` since the keys of the dict are not split names.

Maybe you could check that `sorted(url_or_urls.keys()) == sorted(self._download_config.splits)` before filtering ?

Reply:
Edit:

I meant data like this 
```
{"main_metadata": url_to_main_data, "secondary_metadata": url_to_sec_data, "train": url_train_data, "test": url_test_data}
```

Reply:
Hi @lhoestq. Sorry, I'm not sure of understanding what you mean...

What I am checking here is that the all the keys in `self._download_config.splits` are a subset of the keys in `url_or_urls`.

Reply:
If you pass a dictionary like this:
```
{"main_metadata": url_to_main_data,
"secondary_metadata": url_to_sec_data,
"train": url_train_data,
"test": url_test_data}
```
then only the train and test keys will be kept, which I feel not intuitive.

For example if the users asks to load the "train" split, then the main and secondary metadata won't be downloaded.
You can fix that by keeping all the keys except the splits to ignore

Reply:
Thanks @lhoestq, I understand it now.

Reply:
See my comment below.